lib/explorer/data_frame.ex

```diff
@@ -188,6 +188,17 @@ defmodule Explorer.DataFrame do
       iex> df["class"][3]
       1
 
+
+  When trying to access a column in the dataframe that does not exist, an `ArgumentError` is raised.
+
+      iex> df = Explorer.Datasets.wine()
+      iex> try do
+      ...>   {:ok, df["abcd"]}
+      ...> rescue
+      ...>   _ in ArgumentError -> :error
+      ...> end
+      :error
```

Comment:
```suggestion
      iex> df["abcd"]
      ** (ArgumentError) could not find column name "omg"
```

Reply:
Ah I just assumed the above would work 😬 

Yeah I'd honestly go even simpler:

```elixir
iex> df =  DataFrame.new(a: [1, 2, 3])
iex> df["b"]
** (ArgumentError) could not find column name "b". The available columns are: ["a"].
If you are attempting to interpolate a value, use ^b.
```

^ Not tested...

Reply:
I'm running into an issue matching on this error message:

```
     Doctest failed: wrong message for ArgumentError
     expected:
       "could not find column name \"b\". The available columns are: [\"a\"].\nIf you are attempting to interpolate a value, use ^b."
     actual:
       "could not find column name \"b\". The available columns are: [\"a\"].\nIf you are attempting to interpolate a value, use ^b.\n"
```

It has a trailing newline and I cannot get that to work.  I have tried various combinations of spaces and newlines to try and get it to work out...but I have not been able to.  Turns out that the link I posted previosly (https://elixirforum.com/t/handling-multi-newline-exception-messages-in-doctests-elixir-v1-12-0/39912) is actually talking about the same problem the doctest matching the argument error.  And it looks like a problem isn't addressable right now.

So I'm thinking the best thing is to use the `try/rescue` way after all. What do you think?

Reply:
Hm that's a shame! I didn't know about this issue with doctests. Yeah I think going back to the `try/rescue` is a fine workaround.

BTW I tried this too but I don't like it:

```
      iex> df = Explorer.DataFrame.new(a: [1, 2, 3])
      iex> try do
      ...>   df["b"]
      ...> rescue
      ...>   error in ArgumentError -> error.message
      ...> end
      ~s'could not find column name "b". The available columns are: ["a"].
      If you are attempting to interpolate a value, use ^b.
      '
```

Reply:
We should fix the error message by adding a `\` at the end so it removes the trailing new line.

Reply:
@mooreryan If you want to try adding the `\` on this PR I think that's totally fine. But if changing it here means you have to run around editing a bunch of other doctests, we (you or the team) can address it in another PR.

Reply:
Oh in this case I think you can just remove the final `\n` on 268.

---

Both José and I assumed the newlines came from a docstring like this:

```elixir
"""
could not find column name "#{name}" #{did_you_mean(name, df.names)}
If you are attempting to interpolate a value, use #{name}.
"""
```

When you use docstrings like that, you get a trailing newline. The workaround is to add a `\` as the last character like so:

```elixir
"""
could not find column name "#{name}" #{did_you_mean(name, df.names)}
If you are attempting to interpolate a value, use #{name}.\
"""
```

But here the message was built by gluing strings with newline characters, so our advice wasn't applicable.

Reply:
> @mooreryan If you want to try adding the `\` on this PR I think that's totally fine. But if changing it here means you have to run around editing a bunch of other doctests, we (you or the team) can address it in another PR.

I removed the newline in the message.  It only affected two tests, so it seemed small enough to include in this PR.